### PR TITLE
Fixes #24 but not #13

### LINKED ISSUE #24: Special Transports don't send job accepted
In my ATS installation special transports don't send a request to v1/job with status 1 when accepting. Only when delivering with status 3. Havent tested cancelling

Also the UI shows nothing:
pic

Ingame:

### PR DESCRIPTION
**Pull request by Subtixx**
This will fix the non-sending bug. However it does not fix the missing source/destination.

### CODE REVIEW

File: plugin/src/Logger.cpp [hunk 1]
```diff
@@ -185,6 +188,8 @@ SCSAPI_VOID Logger::configuration(const scs_telemetry_configuration_t *event_inf
         }
 
         bool onJob = (found == 13);
```
Comment (Lihis) [inline plugin/src/Logger.cpp]: Maybe just `bool onJob = (found == (isSpecial ? 9 : 13));`.

Comment (Subtixx) [reply plugin/src/Logger.cpp]: > 
> 
> Finally with a thought looked the SDK relating this issue.
> 
> So it's only company source/destination which is missing. With these changes the GUI will now show for example `Origin: Helsinki ()` when having a special job?

Exactly

**Comment by Subtixx** [comment]
> 
> 
> Finally with a thought looked the SDK relating this issue.
> 
> So it's only company source/destination which is missing. With these changes the GUI will now show for example `Origin: Helsinki ()` when having a special job?

Exactly

**Comment by Lihis** [comment]
In my opinion it looks little off if the there is just `Origin: City ()`. Some logic in `MainWindow.cpp` to show special jobs as `Origin: City` would be nice. Or what do you think?

**Comment by Subtixx** [comment]
Honestly yes, but it was just a quick fix that I have on my local copy because my WebUI was always throwing errors because it doesnt have any open jobs to "close".

Maybe just hide () or show (Special)?

**Comment by Subtixx** [comment]
Changes:
screen

**Comment by donatocl** [comment]
So with this fix, the json post comming from the app to the web-api will send in source/destination company "special"?

**Comment by Subtixx** [comment]
No.

Special jobs now also send to /api/v1/job with status 1 when accepting. The source / destination will be in this case be missing from the JSON file

**Comment by Lihis** [comment]
> No.
> 
> Special jobs now also send to /api/v1/job with status 1 when accepting. The source / destination will in this case be missing from the JSON file

They would not be missing but they would be empty strings?

**Comment by Subtixx** [comment]
> 
> 
> > No.
> > Special jobs now also send to /api/v1/job with status 1 when accepting. The source / destination will in this case be missing from the JSON file
> 
> They would not be missing but they would be empty strings?

Or this 🤣  Havent had any experience using msgpack tbh.

**Comment by donatocl** [comment]
I think it is better to send the value "special" and not an empty one, it is simpler for the web api to identify, is possible?

**Comment by Subtixx** [comment]
Or possibly a new bool value isSpecial. Since without having one in m_Job its impossible to know if its a special job or not in on_job_update

**Comment by Lihis** [comment]
I would go for `isSpecial`. If `true` then `company` in keys would not exist in the JSON.

**Comment by Subtixx** [comment]
Uff. Thats over my head unfortunately. No idea how to do this with msgpack

**Comment by donatocl** [comment]
> 
> 
> I would go for `isSpecial`. If `true` then `company` keys would not exist in the JSON.

Cool, so with this, in the web side, I can do a search inside the array to see if the source->company->id and destination->company->id exists, and if not exists I can determine that is an Special Transport Job.

**Comment by Lihis** [comment]
> Uff. Thats over my head unfortunately. No idea how to do this with msgpack

I should have it done soon.

**Comment by donatocl** [comment]
> 
> 
> > Uff. Thats over my head unfortunately. No idea how to do this with msgpack
> 
> I should have it done soon.

Thank you a lot!

**Comment by Lihis** [comment]
Created PR, review is welcome.

**Pull request merged**